bootstrap.sample.json

```diff
@@ -25,5 +25,34 @@
       "_id": "local",
       "type": "engine"
     }
+  ],
+  "filetypes": [
+    { "_id": "bval", "regex": ".*\\.(bval$|bvals$)" },
```

Comment:
This implementation might identify `x.dcm.zip` as an `archive` (instead of as a `dicom`) depending on the order of iteration over filetypes/regexes. This was working in the previously because we were using an exact match for *after the first dot* [like so](https://github.com/scitran/core/blob/master/api/files.py#L163).

Let's remove `.*` from the beginning of regexes (they get even simpler) and switch to `re.search()` instead of `re.match()`. If multiple matches are found, one could still infer that the best match is the longest one. That way we can still get the added flexibility of regexes (eg. for matching EFiles) but don't lose existing functionality.

Reply:
Nice catch!

I wonder if there's an algorithm that can tell us if a set of regexes are mutually exclusive :thinking: